renovate/create-wrapper.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -euo pipefail
+
+# Script to create a Renovate wrapper that auto-exports RENOVATE_* secrets
+# Usage: create-wrapper.sh <output-path>
+# Note: Expects SECRETS_JSON environment variable to be set
+
+OUTPUT_PATH="$1"
+
+echo "🔧 Creating Renovate wrapper script..."
+
+# Validate that SECRETS_JSON is set
+if [ -z "${SECRETS_JSON:-}" ]; then
+  echo "❌ Error: SECRETS_JSON environment variable is not set"
+  exit 1
+fi
+
+# Clean and validate the JSON using jq (this removes control characters)
+echo "🔍 Cleaning and validating JSON..."
+CLEANED_JSON=$(echo "$SECRETS_JSON" | jq -c . 2>&1) || {
+  echo "❌ Error: Invalid JSON in SECRETS_JSON"
+  exit 1
+}
+
+echo "✅ JSON cleaned and validated"
+
+# Encode to base64 (use -w 0 to disable line wrapping)
+SECRETS_BASE64=$(echo "$CLEANED_JSON" | base64 -w 0)
+
+echo "✅ Secrets validated and encoded"
+
+# Create the wrapper script with base64-encoded secrets
+cat > "$OUTPUT_PATH" <<WRAPPER_EOF
+#!/bin/bash
+set -euo pipefail
+
+# Decode secrets from base64
+ALL_SECRETS=\$(echo "${SECRETS_BASE64}" | base64 -d)
+
+echo "🔧 Auto-detecting and exporting RENOVATE_* secrets..."
+echo ""
+
+# Track count of exported secrets
+EXPORTED_COUNT=0
+
+# Export each RENOVATE_* secret
+while IFS='=' read -r key value; do
+  if [ -n "\$key" ] && [ -n "\$value" ]; then
+    export "\$key=\$value"
+    EXPORTED_COUNT=\$((EXPORTED_COUNT + 1))
+    
+    # Mask sensitive values in output
+    if [[ \$key == *"PASSWORD"* ]] || [[ \$key == *"TOKEN"* ]] || [[ \$key == *"KEY"* ]]; then
+      echo "  ✓ \$key=***"
+    else
+      echo "  ✓ \$key=\$value"
+    fi
+  fi
+done < <(echo "\$ALL_SECRETS" | jq -r 'to_entries[] | select(.key | startswith("RENOVATE_")) | "\(.key)=\(.value)"')
+
+echo ""
+echo "✅ Exported \$EXPORTED_COUNT RENOVATE_* secrets"
+echo "✅ Configuration complete"
+echo ""
+echo "🚀 Starting Renovate..."
+echo ""
+
+exec renovate "\$@"
```

Comment:
Why `exec renovate` and not simply `renovate`? Starting Renovate as a subprocess should just be fine here, but moreover, it would not require you to `export` the secrets. Instead, if they're present as variables then Renovate will just take them as-is.

Reply:
I'll run it without exec, but I don't think there's an issue in using "export" here, since it's only valid for the lifetime of the step it's executing, so will be gone afterwards.

Reply:
Or am i missing something? D: